src/simplewallet/simplewallet.cpp

```diff
@@ -3691,6 +3805,17 @@ bool simple_wallet::show_transfers(const std::vector<std::string> &args_)
     }
   }
 
+  // subaddr_index
+  std::set<uint32_t> subaddr_indices;
+  if (local_args.size() > 0 && local_args[0].substr(0, 6) == "index=")
+  {
+    if (!parse_subaddress_indices(local_args[0], subaddr_indices))
+      return true;
+    in = pool = true;
+    out = pending = failed = false;
```

Comment:
This is done because each incoming payment (obtained via `get_payments` & `get_unconfirmed_payments`) can be uniquely associated with an address index (i.e. a payment cannot be simultaneously received by more than one subaddresses), while each outgoing payment (obtained via `get_payments_out` & `get_unconfirmed_payments`) can be associated with more than one address indices (since funds received by different subaddresses can be combined). Therefore, specifying the argument `index=N` makes sense only for incoming payments.